server/game/gameSystems/TakeControlOfUnitSystem.ts

```diff
@@ -0,0 +1,47 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, GameStateChangeRequired, ZoneName, WildcardCardType, EventName } from '../core/Constants';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
+import Player from '../core/Player';
+
+export interface ITakeControlProperties extends ICardTargetSystemProperties {
+    newController: Player;
+}
+
+/**
+ * Used for taking control of a unit, upgrade, or resource without making any other change.
```

Comment:
The system name implies it is supposed to be limited to units. This comment might be an oversight.